doc/nrf/ug_thread_ot_integration.rst

```diff
@@ -0,0 +1,130 @@
+The drawback, on the other hand, lies in the reception path in which the IP packet needs to traverse the OpenThread L3 layer and then Zephyr's L3 layer in order to reach the BSD socket.
+
+.. note:: If one desires the application can use just the OpenThread API with it's IPv6 stack and even the internal CoAP implementation which is the application protocol.
+          Such approach reduces the overhead, becouse the packet does not need to go through the Zephyr's stack. 
+          This could lead to createing a quicker or more energy efficient application.
+          However it comes at at a cost of limited portability and functionality.
+          Higher level protocols provided by OpenThread are not as advanced comparing to Zephyr's.
+          For example, CoAP implementation does not support observers and the block transfer functionality.
+          Moreover the TCP protocol is not available at all.
+
+File system and shim layer
+**************************
+
+The OpenThread Network Stack is present in the following paths:
+
+* OpenThread stack location: ``modules/lib/openthread/``
+* OpenThread shim layer: 
+  
+  * Thread entry point function, callbacks, utils, L2 registration: ``zephyr/subsys/net/l2/openthread/``
+  * OpenThread platform layer location: ``zephyr/subsys/net/lib/openthread/platform/``
+
+The shim layer responsibilities include:
+
+* translating the data into the Zephyr's native net_pkt,
+* providing openthread thread body and synchronization API,
+* providinng ``openthread_send()``, ``openthread_recv()`` calls which are registered as the L2 interface API,
+* providing a way to initialize the OpenThread stack,
+* implementing callback functions used by the OpenThread stack
+
+The nRF IEEE802.15.4 Radio Driver is present in the following paths:
+
+* nRF IEEE802.15.4 Radio Driver shim layer location: ``zephyr/drivers/ieee802154/{ieee802154_nrf5.c/ieee802154_nrf5.h}``
+* nRF IEEE802.15.4 Radio Driver location: ``modules/hal/nordic/drivers/nrf_radio_802154``
+The RX connection of the radio driver is done with the interrupt handler registered using the Zephyr's mechanism with the ``NRF_802154_INTERNAL_RADIO_IRQ_HANDLING=0`` defined. 
+The registered IRQ handler uses Zephyr's FIFO to pass the IEEE802.15.4 frame further.
+The ``802154 RX`` thread runs on the highest cooperative priority and waits on this FIFO. 
+Once a new frame appears, it continues with the processing.
+
+The TX connection uses the workqueue, which calls the radio driver calls to schedule the transmission.
+Then the RTC IRQ is used to send the frame over the air.
+
+Threads
+*******
+
+* ``openthread`` - responsible for receiving IEEE 802.15.4 frames during reception.
+  When the reassembled traffic turns out to be the IPv6 packet, it calls the ``ot_receive_handler()`` which injects it back to the L2 via the ``net_recv_data()``, so that it could later reach the Zephyr's IP stack.
```

Comment:
```suggestion
  When the reassembled traffic is an IPv6 packet, it calls the ``ot_receive_handler()`` which injects it back to the L2 via the ``net_recv_data()``, so that it could later reach the Zephyr's IP stack.
```